### ISSUE TITLE
Broken dependencies

### ISSUE THREAD

**Issue by jon4hz** [ipfs/go-ipfs#8622]
### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/go-ipfs/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [go-ipfs version](https://dist.ipfs.io/#go-ipfs) or have an issue updating.

### Installation method

built from source

### Version

```Text
Repo version 11
```

### Config

```json
-
```

### Description

I'm trying to run the "go-ipfs-as-a-library" example and it seems like there are some broken dependencies. I upgraded go-ipfs to v0.11.0 in the go.mod file and ran `go mod tidy` but I'm still getting these errors.

```
$ go run main.go
# github.com/libp2p/go-libp2p-transport-upgrader
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/conn.go:16:12: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/conn.go:38:32: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/upgrader.go:71:11: undefined: network.Stat
# github.com/libp2p/go-libp2p-peerstore/pstoremem
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.4.0/pstoremem/keybook.go:19:5: cannot use (*memoryKeyBook)(nil) (type *memoryKeyBook) as type "github.com/libp2p/go-libp2p-core/peerstore".KeyBook in assignment:
        *memoryKeyBook does not implement "github.com/libp2p/go-libp2p-core/peerstore".KeyBook (missing RemovePeer method)
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.4.0/pstoremem/metadata.go:28:5: cannot use (*memoryPeerMetadata)(nil) (type *memoryPeerMetadata) as type "github.com/libp2p/go-libp2p-core/peerstore".PeerMetadata in assignment:
        *memoryPeerMetadata does not implement "github.com/libp2p/go-libp2p-core/peerstore".PeerMetadata (missing RemovePeer method)
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.4.0/pstoremem/peerstore.go:35:3: cannot use "github.com/libp2p/go-libp2p-peerstore".NewMetrics() (type *"github.com/libp2p/go-libp2p-peerstore".metrics) as type "github.com/libp2p/go-libp2p-core/peerstore".Metrics in field value:
        *"github.com/libp2p/go-libp2p-peerstore".metrics does not implement "github.com/libp2p/go-libp2p-core/peerstore".Metrics (missing RemovePeer method)
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.4.0/pstoremem/protobook.go:36:5: cannot use (*memoryProtoBook)(nil) (type *memoryProtoBook) as type "github.com/libp2p/go-libp2p-core/peerstore".ProtoBook in assignment:
        *memoryProtoBook does not implement "github.com/libp2p/go-libp2p-core/peerstore".ProtoBook (missing RemovePeer method)
# github.com/libp2p/go-libp2p-swarm
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm.go:216:11: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm_conn.go:42:7: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm_conn.go:174:23: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm_conn.go:204:10: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm_stream.go:31:7: undefined: network.Stat
/home/jonah/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.8.0/swarm_stream.go:154:25: undefined: network.Stat
```

When I replaced `go-libp2p` with a local clone of the repository, it worked as expected.

**Comment by marten-seemann** [comment]
Please refrain from updating libp2p dependencies deep down in the stack. The set of version released as go-libp2p v0.16.0 works.

**Comment by jon4hz** [comment]
replacing libp2p was just to test if there is really a broken dependency. 
But somehow my other dependencies where messed up, too. However the current go-ipfs-as-a-library example works only if you update the go-ipfs version to v0.11.0.
